sql/core/src/main/scala/org/apache/spark/sql/execution/debug/package.scala

```diff
@@ -107,18 +109,20 @@ package object debug {
   case class DebugExec(child: SparkPlan) extends UnaryExecNode with CodegenSupport {
     def output: Seq[Attribute] = child.output
 
-    class SetAccumulator[T] extends AccumulatorV2[T, HashSet[T]] {
-      private val _set = new HashSet[T]()
+    class SetAccumulator[T] extends AccumulatorV2[T, java.util.Set[T]] {
+      private val _set = Collections.synchronizedSet(new java.util.HashSet[T]())
```

Comment:
If you use `Collections.synchronized*`, will serialization of those objects also be thread-safe (i.e. will `writeObject` synchronize properly)? What about if Kryo is used?